services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/NeighborRepository.java

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.portmanager.repo;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.stats.DurationStatistics;
+import com.futurewei.alcor.portmanager.entity.PortNeighbors;
+import com.futurewei.alcor.web.entity.dataplane.NeighborInfo;
+import com.futurewei.alcor.web.entity.port.PortEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class NeighborRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(NeighborRepository.class);
+    private static final String NEIGHBOR_CACHE_NAME_PREFIX = "neighborCache-";
+
+    private ICache<String, PortNeighbors> neighborCache;
+    private CacheFactory cacheFactory;
+
+    public NeighborRepository(CacheFactory cacheFactory) {
```

Comment:
@chenpiaoping When you get a chance, please update the PM design doc and list the repositories used in Port? It is time for us to sync the doc with the implementation as we are looking into perf optimization.